tenant/middleware_bucket_metrics.go

```diff
@@ -0,0 +1,77 @@
+package tenant
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/kit/metric"
+	"github.com/influxdata/influxdb/kit/prom"
+)
+
+type bucketMetrics struct {
+	// RED metrics
+	rec *metric.REDClient
+
+	bucketService influxdb.BucketService
+}
+
+var _ influxdb.BucketService = (*bucketMetrics)(nil)
+
+// NewBucketMetrics returns a metrics service middleware for the Bucket Service.
+func NewBucketMetrics(reg *prom.Registry) influxdb.BucketMiddleware {
```

Comment:
By using a middleware generic function the call here can get confusing 'NewBucketMetrics(reg).(svc)'